### PR TITLE
engine: update buildkit to commit w/ more cache ref logs

### PR DESCRIPTION
**Pull request by sipsma**
I sent out a PR to buildkit last week for a few more logs that may be useful for debugging on top of my previous update: https://github.com/moby/buildkit/pull/3905

### CODE REVIEW

**Comment by sipsma** [comment]
Think there is a non-flake failure in `TestContainerExport`, probably some backwards incompatible change in buildkit got picked up from a separate commit picked up with the upgrade here, looking into it

**Comment by sipsma** [comment]
Unfortunately the backwards incompatible change that got picked up incidentally is not incredibly simple to fix:
1. Previously, containers exported as tarballs included both oci and docker compatible manifests ([as verified in our tests here](https://github.com/sipsma/dagger/blob/eefb8f9b405e997695656318c4a70667301699eb/core/integration/container_test.go#L2706-L2708))
   * [Our docs rely on the docker compatibility feature here](https://docs.dagger.io/252029/load-images-local-docker-engine/#approach-1-use-an-exported-tarball)
2. Now, exported images will only have oci manifests by default, which seems to be a result of [this change in containerd](https://github.com/containerd/containerd/pull/8213) 
   * [Some discussion in buildkit too here PR](https://github.com/moby/buildkit/issues/1949#issuecomment-1453244801)

This means that unless we change something, this update will also result in backwards incompatibility.

I have a guess at an approach that might work with changes to our code only:
1. If a multiplatform image is being exported, use `ExporterOCI`. `manifest.json` doesn't support multi-platform images anyways
2. If a single platform image is being exported, use `ExporterDocker`. [Based on some buildkit tests](https://github.com/sipsma/buildkit/blob/60d134bf7671e60bf2a2b9bc41a1d856e6ef396e/client/client_test.go#L2444-L2444), it looks like even those will still have oci indexes, so they should be oci compatible too?

Have to pivot to something else atm, will try this later tonight.

**Comment by sipsma** [comment]
I think my update [as described here](https://github.com/dagger/dagger/pull/5223#issuecomment-1569286964) worked, at least it seems that all the existing tests pass, except for one slight modification which was only needed because a bug in the previous implementation was fixed (estargz compression tried to use a nil `Attrs` map).

cc @vito when you have time let me know if you think this update makes sense too. I think we should add optional args to our publish/export apis that allow users to force use of oci or docker types, but I really don't want to expand the scope of this PR, which was supposed to just be a `go.mod` version bump, out into new API features 😅 So this just attempts to retain the previous behavior to the extent possible.

**Comment by vito** [comment]
@sipsma Having gone over the linked discussion (thanks for finding it!), I think your change makes the most sense.

I think it could make sense to add a `format: OCI | Docker` API, but I would prefer to wait until someone really needs it. It would be nice to end up with pure-OCI in the long run, which we might be getting closer to (https://github.com/moby/moby/issues/25779 was closed 2 weeks ago!).

**Comment by sipsma** [comment]
> I think it could make sense to add a `format: OCI | Docker` API, but I would prefer to wait until someone really needs it. It would be nice to end up with pure-OCI in the long run, which we might be getting closer to ([moby/moby#25779](https://github.com/moby/moby/issues/25779) was closed 2 weeks ago!).

Oh good find, I totally agree about not rushing into that. We'll have to wait for the new docker release to make it's way out in the world's install base, but it would be very nice if that happens before anyone needs support for something besides OCI.

File: core/integration/container_test.go
Comment (carlos2martinize) [inline core/integration/container_test.go:1]: Moniture repository directory

**Pull request merged**